code/WorkInProgress/MechanicMadness.dm

```diff
@@ -2413,16 +2413,22 @@
 		frequency = new_frequency
 		radio_connection = radio_controller.add_object(src, "[frequency]")
 		tooltip_rebuild = 1
-	proc/hear_radio(mob/M as mob, msg, lang_id)
+	proc/hear_radio(atom/movable/AM as mob|obj, msg, lang_id)
 		if (level == 2) return
 		LIGHT_UP_HOUSING
 		var/message = msg[2]
 		if (lang_id in list("english", ""))
 			message = msg[1]
 		message = strip_html(html_decode(message))
-		var/heardname = M.real_name
```

Comment:
Why would you put a : in the replacement of this? It'll runtime if you pass an obj...

Reply:
`obj` has `real_name` as well

Reply:
actually I'm going to change this anyway since `name` makes more sense for objects